File: Corte1/Taller1/Taller.py
import tkinter as tk
from tkinter import simpledialog, messagebox
import time

def ejecutar():
    start_time = time.perf_counter()  # Tiempo inicial


    root = tk.Tk()
    root.withdraw()

    base = None
    while base is None or base == 0:
        base_str = simpledialog.askstring("Ventana", "Primer número (base): ")
        if base_str is not None:
            base = int(base_str)
            if base == 0:
                messagebox.showerror("Error", "El primer número no puede ser cero. Ingrese de nuevo.")
        else:
            return  

    exponente_str = simpledialog.askstring("Ventana", "Segundo número (exponente): ")
    if exponente_str is None:
        return 
    exponente = int(exponente_str)


    resultados = []
    for i in range(exponente + 1):
        resultado = potenciar(base, i)
        resultados.append(f"{base}^{i} = {resultado}")

    end_time = time.perf_counter()  # Tiempo final
    duration_ns = (end_time - start_time) * 1e9  # Duración en nanosegundos

    resultados.append(f"\nTiempo de ejecución: {duration_ns:.0f} nanosegundos")
    messagebox.showinfo("Resultados", "\n".join(resultados))

def potenciar(base, exponente):
    resultado = 1
    for i in range(1, exponente + 1):
        resultado *= base
    return resultado

if __name__ == "__main__":
    ejecutar()

File: Corte1/Taller1/Taller.js
const { app, BrowserWindow, dialog } = require('electron');
const { performance } = require('perf_hooks');
const prompt = require('electron-prompt'); // Importar electron-prompt

// Función para potenciar
function potenciar(base, exponente) {
    let resultado = 1;
    for (let i = 1; i <= exponente; i++) {
        resultado *= base;
    }
    return resultado;
}

// Función para solicitar entradas y mostrar resultados
async function ejecutar() {
    try {
        let tiempoInicio = performance.now();

        // Solicitar base
        let base = parseInt(await prompt({
            title: 'Entrada de Datos',
            label: 'Primer número (base):',
            type: 'input',
            inputAttrs: {
                type: 'text'
            }
        }), 10);

        while (base === 0 || isNaN(base)) {
            await dialog.showErrorBox('Error', 'El primer número no puede ser cero. Ingrese de nuevo.');
            base = parseInt(await prompt({
                title: 'Entrada de Datos',
                label: 'Ingrese de nuevo el primer número (base):',
                type: 'input',
                inputAttrs: {
                    type: 'text'
                }
            }), 10);
        }

        // Solicitar exponente
        let exponente = parseInt(await prompt({
            title: 'Entrada de Datos',
            label: 'Segundo número (exponente):',
            type: 'input',
            inputAttrs: {
                type: 'text'
            }
        }), 10);

        // Procesar datos
        let resultados = '';

        for (let i = 0; i <= exponente; i++) {
            let resultado = potenciar(base, i);
            resultados += `${base}^${i} = ${resultado}\n`;
        }

        let tiempoFin = performance.now();
        let tiempoEjecucion = (tiempoFin - tiempoInicio) * 1e6; // Convertir a nanosegundos

        // Mostrar resultados
        await dialog.showMessageBox({
            type: 'info',
            title: 'Resultados',
            message: `${resultados}\nTiempo de ejecución: ${tiempoEjecucion.toFixed(0)} nanosegundos`
        });
    } catch (err) {
        console.error('Error durante la ejecución:', err);
    } finally {
        app.quit(); // Cerrar la aplicación después de mostrar los resultados
    }
}

app.whenReady().then(() => {
    // Crear una ventana oculta
    const ventana = new BrowserWindow({
        show: false, // Ocultar la ventana
        webPreferences: {
            nodeIntegration: true,
            contextIsolation: false
        }
    });

    ventana.loadURL('data:text/html,<html><body></body></html>'); // Cargar una página en blanco para evitar errores

    // Ejecutar la función principal
    ejecutar();
});

app.on('window-all-closed', () => {
    if (process.platform !== 'darwin') {
        app.quit();
    }
});
